src/couch_epi_functions_gen.erl

```diff
@@ -61,6 +63,23 @@ apply(Handle, _ServiceId, Function, Args, Opts) ->
     Modules = providers(Handle, Function, length(Args), DispatchOpts),
     dispatch(Handle, Modules, Function, Args, DispatchOpts).
 
+-spec defined_by_app(
+    Handle :: atom(), SourceApp :: atom(),
+    Function :: atom(), Arity :: pos_integer()) ->
+        list().
+defined_by_app(Handle, SourceApp, Function, Arity) ->
+    try
+        ModulesBySource = [
+            M || {M, _} <- get_current_definitions(Handle, SourceApp)
+        ],
+        ModulesByFunc = Handle:providers(Function, Arity),
+        lists:filter(fun(I) ->
+            lists:member(I, ModulesBySource)
+        end, ModulesByFunc)
+    catch
+        error:undef -> false
```

Comment:
Also, I think you have here a bug: if undef error raised, in `is_defined_by_app/4` you'll compare `[] /= false` which give you `true`, but don't think that's right answer,